tests/test_eventgrid_functions.py

```diff
@@ -0,0 +1,66 @@
+import json
+import time
+import requests
+
+from azure.worker import testutils
+
+
+class TestEventGridFunctions(testutils.WebHostTestCase):
+
+    @classmethod
+    def get_script_dir(cls):
+        return 'eventgrid_functions'
+
+    def request(self, meth, funcname, *args, **kwargs):
+        request_method = getattr(requests, meth.lower())
+        url = (f'{self.webhost._addr}/admin/extensions/'
+               f'EventGridExtensionConfig?functionName={funcname}')
+        print(url)
+        return request_method(url, *args, **kwargs)
+
+    def test_eventgrid_trigger(self):
+        data = [{
+            "topic": "/subscriptions/{subscriptionid)/resourceGroups/"
+                     "eg0122/providers/Microsoft.Storage/storageAccounts"
+                     "/egblobstor0122",
+            "subject": "/blobServices/default/containers/test0123/"
+                       "blobs/Default.rdp",
+            "eventType": "Microsoft.Storage.BlobCreated",
+            "eventTime": "2018-01-23717:02:19.6069787Z",
+            "id": "44d4f022-001e-003c-466b-940cba0612e5",
+            "data": {
+                "api": "PutBlockList",
+                "clientRequestId": "2c169f2f-7b3b-4d99-839b-c92a2d25801b",
+                "requestId": "44d4f022-001e-003c-466b-940cba000000",
+                "eTag": "0x8D562831044DDD0",
+                "contentType": "application/octet-stream",
+                "contentLength": 2248,
+                "blobType": "BlockBlob",
+                "ur1": "https://{storageaccountname).blob.core.windows.net/test0123/Default.rdp",
+                "sequencer": "000000000000272D000000000003D60F",
+                "storageDiagnostics": {
+                    "batchId": "b4229b3a-4d50-4ff4-a9f2-039ccf26efe9"
+                }
+            },
+            "dataVersion": "",
+            "metadataVersion": "1"
+        }]
+
+        r = self.request('POST', 'eventgrid_trigger', json=data)
+        self.assertEqual(r.status_code, 200)
```

Comment:
From the doucmentation on how to [manually-post-the-request](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-event-grid#manually-post-the-request), you need Set an aeg-event-type: Notification header.